fieldservice_sale/models/fsm_order.py

```diff
@@ -20,3 +20,15 @@ def action_view_sales(self):
             "context": {"create": False},
             "name": _("Sales Orders"),
         }
+
+    def write(self, vals):
+        for order in self:
+            if "customer_id" not in vals and not order.customer_id:
```

Comment:
I see... 
Since the field is defined in fieldservice_accont_analytic, how would you consider the idea to add that module as a dependency?
Otherwise all this PR would loose meaning

Let me know if there is any different approach you would've preferred

Reply:
There is different approach here. The most two obvious are: 

-  you use fieldservice_sale then you may want to invoice based on the sale order (and use analytics from sale order lines)
-  you use create orders directly so you don't have a sale, thanks to fieldservice_account_analytic,  you invoice the fsm orders directly.


Do you use another approach ? 

From a technical perspective, you can put your code in a new module like `fieldservice_sale_analytic`.

Reply:
We create the sale order and let the fsm order be generated by `fieldservice_sale` . 
The problem is the `customer_id` that results empty when it should be the same customer of the sale order 

This means that `customer_id` should always be available to `fieldervice_sale` since otherwise it breaks the module and the options are re-difining `customer_id` or add the dependency of `fieldservice_sale_analytic` . 

I see this as a bug since otherwise the `customer_id` is not automatically set. 
Let me know your thoughts about it and how you think is best to proceed, or even if it's to be considered a bug at all!

Reply:
> From a technical perspective, you can put your code in a new module like `fieldservice_sale_analytic`.

do you think that for this approach it needs to be created a new module?

Reply:
If you use sale to generate the fsm, you do not need customer_id. The order will be invoiced directly by the sale.